src/plugins/matrixops/min_operation.cpp

```diff
@@ -0,0 +1,284 @@
+        }
+
+        return primitive_argument_type{arg.scalar()};
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min1d(ir::node_data<T>&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis && axis.value() != 0 && axis.value() != -1)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "min_operation::min1d",
+                generate_error_message(
+                    "the min_operation primitive requires operand axis to be "
+                    "either 0 or -1 for vectors."));
+        }
+        auto v = arg.vector();
+        T result = blaze::min BOOST_PREVENT_MACRO_SUBSTITUTION(v);
+
+        if (keep_dims)
+        {
+            return primitive_argument_type{blaze::DynamicVector<T>{result}};
+        }
+        return primitive_argument_type{ir::node_data<T>{result}};
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min2d(ir::node_data<T>&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis)
+        {
+            switch (axis.value())
+            {
+            case -2:
+                HPX_FALLTHROUGH;
+            case 0:
+                return min2d_axis0(std::move(arg), keep_dims);
+
+            case -1:
+                HPX_FALLTHROUGH;
+            case 1:
+                return min2d_axis1(std::move(arg), keep_dims);
+
+            default:
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "min_operation::min2d",
+                    generate_error_message(
+                        "the min_operation primitive requires operand axis "
+                        "to be between -2 and 1 for matrices."));
+            }
+        }
+        return min2d_flat(std::move(arg), keep_dims);
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min2d_flat(
+        ir::node_data<T>&& arg, bool keep_dims) const
+    {
+        auto m = arg.matrix();
+        T result = blaze::min BOOST_PREVENT_MACRO_SUBSTITUTION(m);
+
+        if (keep_dims)
+        {
+            return primitive_argument_type{blaze::DynamicMatrix<T>{{result}}};
+        }
+        return primitive_argument_type{ir::node_data<T>{result}};
+    }
+
+    template <typename T>
+    primitive_argument_type min_operation::min2d_axis0(
+        ir::node_data<T>&& arg, bool keep_dims) const
+    {
+        auto m = arg.matrix();
+        blaze::DynamicVector<T> result(m.columns());
+        for (std::size_t i = 0; i < m.columns(); ++i)
+        {
+            result[i] =
+                blaze::min BOOST_PREVENT_MACRO_SUBSTITUTION(column(m, i));
+        }
+
+        if (keep_dims)
+        {
+            return primitive_argument_type{
+                blaze::DynamicMatrix<T>{1, result.size(), result.data()}};
+        }
+        return primitive_argument_type{ir::node_data<T>{result}};
```

Comment:
You might want to move the result into the returned `primitive_argument_type` instance.